core-ui/src/resources/helpers.js

```diff
@@ -0,0 +1,48 @@
+import { useTranslation } from 'react-i18next';
+import { useParams } from 'react-router-dom';
+import { useMicrofrontendContext } from 'shared/contexts/MicrofrontendContext';
+import { getResourceUrl } from 'shared/helpers';
+import { getPerResourceDefs } from 'shared/helpers/getResourceDefs';
+
+export const usePrepareListProps = (resourceType, resourceI18Key) => {
+  const routerParams = useParams();
+  const queryParams = new URLSearchParams(window.location.search);
+  const { i18n, t } = useTranslation();
+  const resourceUrl = getResourceUrl();
+  return {
+    hasDetailsView: queryParams.get('hasDetailsView') === 'true',
+    readOnly: queryParams.get('readOnly') === 'true',
+    resourceUrl,
+    resourceType: resourceType,
+    resourceName: t(resourceI18Key || '') ? t(resourceI18Key) : resourceI18Key,
```

Comment:
`t(resourceI18Key)` will return `resourceI18Key` by default if no transaltion
```suggestion
    resourceName: t(resourceI18Key || ''),
```